src/components/PageLayout/Navbar/Icons/NotificationIcon/NotificationIcon.vue

```diff
@@ -87,15 +94,22 @@
     }
 
     async onShown() {
-      await this.$http.post('/post/', new URLSearchParams({ type: 'notification_read' }));
       await this.$fetch();
     }
 
     async fetch() {
       const res = await this.$http.get(NUM_NOTIFICATIONS_ROUTE);
       this.notificationsCount = res.data.data.noti_count;
-      if (res.data.data.noti_count > this.notificationsCount || !this.fetchState.firstFetchComplete) {
+
+      if (this.notificationsCount > 0 || !this.fetchState.firstFetchComplete) {
+        if ((this.$refs.dropdown as NavbarIcon).isShown) {
+          this.notificationsCount = 0;
+          await this.$http.post('/post/', new URLSearchParams({ type: 'notification_read' }));
+        }
```

Comment:
Maybe this should be done after dropdown contents are updated? Otherwise the user may have opened the menu, but not actually seen the new notifications

Reply:
True, it is fixed now